src/Elastic.Apm.AspNetFullFramework/ElasticApmModule.cs

```diff
@@ -57,22 +58,14 @@ private static void SetServiceInformation(Service service, string aspNetVersion)
 			service.Language = new Language { Name = "C#" }; //TODO
 		}
 
-		public void Init(HttpApplication httpApp)
+		public void Init(HttpApplication context)
 		{
-			_httpApp = httpApp;
-			_httpApp.BeginRequest += OnBeginRequest;
-			_httpApp.EndRequest += OnEndRequest;
+			_context = context;
+			_context.BeginRequest += OnBeginRequest;
+			_context.EndRequest += OnEndRequest;
 		}
 
-		public void Dispose()
-		{
-			if (_httpApp != null)
-			{
-				_httpApp.BeginRequest -= OnBeginRequest;
-				_httpApp.EndRequest -= OnEndRequest;
-				_httpApp = null;
-			}
-		}
+		public void Dispose() => _context = null;
```

Comment:
Shouldn't we unsubscribe from the `BeginRequest` and `EndRequest` events?

Reply:
No. You're using the `HttpApplication` initialization pipeline to bind to them (via `Init`). `Dispose` is outside of the scope `HttpApplication`, so unsubscribing from them in this `Dispose` method is dangerous as it can cause them to be called after being unsubscribed from. This caused crashes on our side. Allow the framework to take care of them being unsubscribed. You might find this [SO question](https://stackoverflow.com/q/2781346/197591) useful.